swarm/src/connection.rs

```diff
@@ -482,6 +461,52 @@ fn gather_supported_protocols(handler: &impl ConnectionHandler) -> HashSet<Strea
         .collect()
 }
 
+fn handle_should_keep_alive(
+    handler: &impl ConnectionHandler,
+    shutdown: &mut Shutdown,
+    idle_timeout: &mut Duration,
+) {
```

Comment:
I think this function would be easier to understand if it would have the following signature:

```suggestion
fn compute_new_shutdown(
    handler: &impl ConnectionHandler,
    current_shutdown: &Shutdown,
    idle_timeout: Duration,
) -> Shutdown {
```

Reply:
rename fn and return new Shutdown instance 412aa75